app/views/flexible_criteria/_criterion_editor.html.erb

```diff
@@ -2,12 +2,14 @@
 
 <div class='float_right'>
   <%= button_to t(:delete),
-      assignment_flexible_criterion_path(id: criterion.id),
-      data: { confirm: t(:remove_criteria_message) },
-      method: :delete,
-      class: 'delete',
-      title: t(:remove_criteria_title),
-      remote: true %>
+    assignment_criterion_path(assignment_id: criterion.assignment.id,
```

Comment:
These lines should be aligned with the `t(:delete)` above, as they are all arguments to `button_to`